enhancements/authentication/service-ca-bundle-injection.md

```diff
@@ -0,0 +1,205 @@
+replaces:
+superseded-by:
+---
+
+# Support Service CA Bundle Injection
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [x] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+A service that needs to securely connect to another service protected
+with a service serving cert (provided by the service CA operator)
+needs to validate that cert with the service CA bundle. Supporting the
+injection of this CA bundle into a number of different resource types
+is intended to minimize the amount of complexity involved in sourcing
+the CA bundle and configuring its use for a variety of use cases.
+
+### Goals
+
+- Service CA bundle injection should be supported for the following
+  types:
+  - `APIService`
+  - `ConfigMap`
+  - `CustomResourceDefinition`
+  - `MutatingWebhookConfiguration`
+  - `ValidatingWebhookConfiguration`
+
+### Non-Goals
+
+- Supporting CA bundle injection to a subset of webhooks defined in an
+  admission webhook configuration resource.
+  - Allowing selective injection would likely increase the complexity
+    of implementation and there is no clear indication that this
+    capability is required.
+  - Webhooks in one configuration object are all independent and
+    therefore configuration can be split into multiple resources if
+    difference CAs are necessary.
+
+## Proposal
+
+- Add controllers to inject the service CA bundle into supported
+  resource types in response to the presence of an injection
+  annotation.
+  - Annotations;
+    - alpha: `service.alpha.openshift.io/inject-cabundle`
+    - beta:  `service.beta.openshift.io/inject-cabundle`
+  - Older types may support both alpha and beta annotations.
+  - Newer types added in 4.4 and above should only respond to the beta
+    annotation.
+
+- Supported resource types:
+  - `APIService` (`apiregistration.k8s.io`)
+    - Injection field: `spec.caBundle`
+    - Supported annotations: alpha,beta
+  - `ConfigMap`
+    - Injection field: `data["service-ca.crt"]`
+    - Supported annotations: alpha,beta
+  - `CustomResourceDefinition` (`apiextensions.k8s.io`)
+    - Injection field: `spec.conversion.webhook.clientConfig.caBundle`
+    - Supported annotations: beta
+    - Notes: Only inject if conversion with webhook strategy is configured
+  - `MutatingWebhookConfiguration` (`admissionregistration.k8s.io`)
+    - Injection field: `webhooks[].clientConfig.caBundle`
+    - Supported annotations: beta
+    - Notes: See below
+  - `ValidatingWebhookConfiguration` (`admissionregistration.k8s.io`)
+    - Injection field: `webhooks[].clientConfig.caBundle`
+    - Supported annotations: beta
+    - Notes: See below
+
+- Injection for `{Mutating|Validating}WebhookConfiguration` resources
```

Comment:
nit: sub-section on Mutating and Validating WebhookConfiguration

Reply:
I'm not sure what you're asking for, can you elaborate?